lib/cpp/prox/prox_sorted_l1.cpp

```diff
@@ -2,21 +2,21 @@
 
 #include "tick/prox/prox_sorted_l1.h"
 
-template <class T>
-void TProxSortedL1<T>::compute_weights(void) {
+template <class T, class K>
+void TProxSortedL1<T, K>::compute_weights(void) {
   TICK_CLASS_DOES_NOT_IMPLEMENT(get_class_name());
 }
 
-template <class T>
-void TProxSortedL1<T>::call(const Array<T> &coeffs, T t, Array<T> &out,
-                            ulong start, ulong end) {
+template <class T, class K>
+void TProxSortedL1<T, K>::call(const Array<K> &coeffs, T t, Array<K> &out,
+                               ulong start, ulong end) {
   // If necessary, compute weights
   compute_weights();
   ulong size = end - start;
   T thresh = t;
 
-  Array<T> sub_coeffs = view(coeffs, start, end);
-  Array<T> sub_out = view(out, start, end);
+  auto sub_coeffs = view(coeffs, start, end);
```

Comment:
Why auto and not `Array<K>`? I think it would be clearer to explicitly tell what type this is as we are constantly switching between `Array<T>` and `Array<K>`.

Reply:
auto is cheaper, and if it ever changes from not K we don't need to update it